### ISSUE TITLE
Enabling both Jackson and JAX-RS modules fails the build

### ISSUE THREAD

**Issue by Biazhkou** [stoicflame/enunciate#984]
A project has many subprojects. Some of them work fine but with another one I have an issue regardless the same configuration.

The subproject are written using JAX-RS annotations so I have enabled **jax-rs module**. 
Jackson module is disabled so in documentation inside request or response body will be represented by xml unless the method is annotated with **@Consumes(MediaType.APPLICATION_JSON)** or **@Produces(MediaType.APPLICATION_JSON)** accordingly. In this case we will see the empty body.

A request example when jackson is disabled and method without @Consumes annotation:

```
POST /url
Content-Type: application/xml
Accept: application/xml

                
<h----->
  <id>...</id>
  <name>...</name>
</h----->
```
It didn't display instance name, only the first letter but it isn't the main issue.

The same when @Consumes json:

```
POST /url
Content-Type: application/json
Accept: application/xml

                
...
```

Looks like I just need to enable **jackson module**. After this action build is failed.

**Configuration** file is the following:
```
<enunciate xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
           xsi:noNamespaceSchemaLocation="http://enunciate.webcohesion.com/schemas/enunciate-2.12.0.xsd">
    <title>...</title>
    <description>
       ...
    </description>
    <copyright>...</copyright>
    <application root="..."/>
    <modules disabledByDefault="true">
        <!-- provider modules -->
        <jaxrs disabled="false"/>
        <jaxb disabled="false"/>
        <jackson disabled="false"/>
        <!-- non provider modules -->
        <docs disabled="false"/>
    </modules>
</enunciate>
```

The **stacktrace** is the following:

```
* What went wrong:
Execution failed for task ':BrooklynApiServices:enunciate'.
> Failed to invoke Enunciate

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':BrooklynApiServices:enunciate'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:103)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:73)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:623)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:578)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: java.lang.IllegalStateException: Failed to invoke Enunciate
        at com.webcohesion.enunciate.gradle.EnunciateTask.run(EnunciateTask.java:173)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:788)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:755)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:124)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:113)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:95)
        ... 30 more
Caused by: java.lang.RuntimeException: java.lang.NullPointerException
        at com.sun.tools.javac.main.Main.compile(Main.java:553)
        at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
        at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
        at com.webcohesion.enunciate.Enunciate.run(Enunciate.java:667)
        at com.webcohesion.enunciate.gradle.EnunciateTask.configureAndInvokeEnunciate(EnunciateTask.java:211)
        at com.webcohesion.enunciate.gradle.EnunciateTask.run(EnunciateTask.java:171)
        ... 43 more
Caused by: java.lang.NullPointerException
        at com.webcohesion.enunciate.modules.jackson.model.TypeDefinition.<init>(TypeDefinition.java:129)
        at com.webcohesion.enunciate.modules.jackson.model.ObjectTypeDefinition.<init>(ObjectTypeDefinition.java:38)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.createTypeDefinition(EnunciateJacksonContext.java:292)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitDeclared(EnunciateJacksonContext.java:661)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitDeclared(EnunciateJacksonContext.java:628)
        at com.webcohesion.enunciate.modules.jackson.javac.ParameterizedJacksonDeclaredType.accept(ParameterizedJacksonDeclaredType.java:51)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.addReferencedTypeDefinitions(EnunciateJacksonContext.java:487)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.add(EnunciateJacksonContext.java:412)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitDeclared(EnunciateJacksonContext.java:661)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitDeclared(EnunciateJacksonContext.java:628)
        at com.webcohesion.enunciate.modules.jackson.javac.ParameterizedJacksonDeclaredType.accept(ParameterizedJacksonDeclaredType.java:51)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitArray(EnunciateJacksonContext.java:632)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitArray(EnunciateJacksonContext.java:628)
        at com.webcohesion.enunciate.javac.decorations.type.DecoratedArrayType.accept(DecoratedArrayType.java:45)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.addReferencedTypeDefinitions(EnunciateJacksonContext.java:487)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.addReferencedTypeDefinitions(EnunciateJacksonContext.java:472)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.add(EnunciateJacksonContext.java:401)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitDeclared(EnunciateJacksonContext.java:661)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitDeclared(EnunciateJacksonContext.java:628)
        at com.webcohesion.enunciate.javac.decorations.type.DecoratedDeclaredType.accept(DecoratedDeclaredType.java:84)
        at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.addReferencedTypeDefinitions(EnunciateJacksonContext.java:487)
        at com.webcohesion.enunciate.modules.jackson.JacksonModule.addDataTypeDefinitions(JacksonModule.java:206)
        at com.webcohesion.enunciate.modules.jaxrs.JaxrsModule.addReferencedDataTypeDefinitions(JaxrsModule.java:232)
        at com.webcohesion.enunciate.modules.jaxrs.JaxrsModule.call(JaxrsModule.java:148)
        at com.webcohesion.enunciate.Enunciate.invokeModules(Enunciate.java:890)
        at com.webcohesion.enunciate.EnunciateAnnotationProcessor.process(EnunciateAnnotationProcessor.java:127)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:705)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
        at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
        at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
        at com.sun.tools.javac.main.Main.compile(Main.java:523)
        ... 48 more
```
Smaller projects works as expected without issues.
Could you advice the way how to resolve it?

**Comment by stoicflame** [comment]
Thanks for the report. Moved to the Enunciate project.

**Comment by Biazhkou** [comment]
It was related to serialization into JSON
We have something like that: ArrayList<SomeType<AnotherType>>
debug said nothing
stacktrace showed internal calls without any specifics

**Comment by stoicflame** [comment]
Fix has been included in Enunciate 2.12.1.
